pkg/bloomgateway/bloomgateway.go

```diff
@@ -383,8 +384,10 @@ func (g *Gateway) consumeTask(ctx context.Context, task Task, tasksCh chan<- Tas
 		case <-ctx.Done():
 			// do nothing
 		default:
-			// chunks may not be sorted
-			sort.Slice(res.Removals, func(i, j int) bool { return res.Removals[i].Less(res.Removals[j]) })
+			// chunks may not always be sorted
+			if !slices.IsSortedFunc(res.Removals, func(a, b v1.ChunkRef) int { return a.Cmp(b) }) {
```

Comment:
nit: we end up using the same cmp funtion all over the places. Would be nice to have another function next to `func (r *ChunkRef) Cmp(other ChunkRef) int` that does this:
```go
func CmpChunkRefs(a, b ChunkRef) int { return a.cmp(b) }
``` 

So we can use IsSortedFunc and SortFunc as:
```
                        if !slices.IsSortedFunc(a.Refs, CmpChunkRefs) {
				slices.SortFunc(a.Refs, CmpChunkRefs)
			}
			if !slices.IsSortedFunc(b.Refs, CmpChunkRefs) {
				slices.SortFunc(b.Refs, CmpChunkRefs)
			}
```